client/js/widgets/scoreboard.js

```diff
@@ -98,31 +146,53 @@ class Scoreboard extends Widget {
       return null;
   }
 
+  // Compute number of scoring rounds to show and create round names table
+  getRounds(seats, scoreProperty, addEmptyRounds=0) {
+    let rounds = this.get('rounds'); // User-supplied round names
+    let numRounds=0;
+    const arrayOfSeats = Array.isArray(seats) ? seats : Object.keys(seats).reduce((union,key) => union.concat(seats[key]), []);
+    for (let i=0; i < arrayOfSeats.length; i++) {
+      const score = arrayOfSeats[i].get(scoreProperty);
+      if(Array.isArray(score) && score.length > numRounds)
+        numRounds = score.length;
+    }
+    if(this.get('showAllRounds') && Array.isArray(rounds))
+      numRounds = Math.max(rounds.length, numRounds);
+    else
+      numRounds += addEmptyRounds;
```

Comment:
If the rounds have been named and all have been used, this will add an empty round named `''`. Is that what we want?